hack/jenkins/e2e.sh

```diff
@@ -225,9 +225,16 @@ fi
 # TODO(zml): We have a bunch of legacy Jenkins configs that are
 # expecting junit*.xml to be in ${WORKSPACE} root and it's Friday
 # afternoon, so just put the junit report where it's expected.
-for junit in ${ARTIFACTS}/junit*.xml; do
-  ln -s ${junit} ${WORKSPACE}
-done
+results=($(ls "${ARTIFACTS}" | grep "junit.*.xml")) || true
+if [[ ! -z ${results:-} ]]; then
+    for junit in "${results[@]}"; do
+        # Jenkins runs that trigger sequentially (e.g. step1 -> step2 -> ...) will
+        # have e2e test results overwrite each other because they all run in the
+        # same workspace. For those cases, we prefix the final location with the
+        # step name.
+        mv "${ARTIFACTS}/${junit}" "${WORKSPACE}/${JENKINS_STEP_PREFIX:-}${junit}"
```

Comment:
1. GCS = Google Cloud Storage. At least several of the Jenkins projects are configured to copy everything under `_artifacts` there so they're viewable outside corp.
2. Your change already uses an additional env var (`$JENKINS_STEP_PREFIX`), so it seems using that or something similar to create a different artifacts directory or junit files should be OK?
3. I think long-term we don't want the junit files in the root dir at all. They should only be in `_artifacts`.